### PR TITLE
Implementation for indexers operations 

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/10836

### CODE REVIEW

File: sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+                :language: python
+                :dedent: 4
+                :caption: List all the Indexers
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        result = self._client.indexers.list(**kwargs)
+        return result.indexers
+
+    @distributed_trace
+    def delete_indexer(self, indexer, **kwargs):
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py:144]: Why we ask user to give entire indexer? Name is good enough. Not correct?

Comment (xiangyan99) [reply sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py:144]: And you have ':param name: The name of the indexer to delete.' :)

Comment (rakshith91) [reply sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py:144]: indexers is `Union[str, Indexer]` - name is enough to delete unconditionally - we need the model to use match conditions - have documented it in the docstrings

File: sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+    def close(self):
+        # type: () -> None
+        """Close the :class:`~azure.search.documents.SearchIndexersClient` session.
+
+        """
+        return self._client.close()
+
+    @distributed_trace
+    def create_indexer(self, indexer, **kwargs):
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py:57]: It returns dict?

File: sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+    @distributed_trace
+    def delete_indexer(self, indexer, **kwargs):
+        # type: (Union[str, Indexer], **Any) -> None
+        """Deletes an indexer. To use only_if_unchanged, the Indexer model
+        must be provided instead of the name. It is enough to provide
+        the name of the indexer to delete unconditionally.
+
+        :param name: The name of the indexer to delete.
+        :type name: str
```
Comment (rakshith91) [inline sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py:151]:
```suggestion
        :param indexer: The indexer to delete.
        :type name: str or ~search.models.Indexer
```

File: sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+                :caption: List all the Indexers
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        result = self._client.indexers.list(**kwargs)
+        return result.indexers
+
+    @distributed_trace
+    def delete_indexer(self, indexer, **kwargs):
+        # type: (Union[str, Indexer], **Any) -> None
+        """Deletes an indexer. To use only_if_unchanged, the Indexer model
```
Comment (rakshith91) [inline sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py:146]:
```suggestion
        """Deletes an indexer. To use access conditions, the Indexer model
```

Comment (xiangyan99) [reply sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py:146]: Have we had an API review?

Comment (bryevdv) [reply sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py:146]: Not yet for Indexers operations, tho if you are asking about `only_if_unchanged` vs access conditions, @johanste  weighed in on that in #11116

File: sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py (hunk 2 above)
Comment (rakshith91) [reply sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py:57]: changed the return type

File: sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py (hunk 4 above)
Comment (rakshith91) [reply sdk/search/azure-search-documents/azure/search/documents/_service/_indexers_client.py]: changed the documentation to match conditions

**Pull request merged**